src/main/java/com/semmle/jira/addon/util/JiraUtils.java

```diff
@@ -43,16 +45,25 @@ public static IssueType getIssueTypeByName(String issueTypeName) {
                 ConstantsManager.CONSTANT_TYPE.ISSUE_TYPE.getType(), issueTypeName);
   }
 
+  public static IssueType getLgtmIssueType() {
+    return getIssueTypeByName(Constants.ISSUE_TYPE_NAME);
+  }
+
   public static void addWorkflowToProject(
-      Project project, JiraWorkflow workflow, IssueType issueType) throws GenericEntityException {
+      Project project, JiraWorkflow workflow, IssueType issueType)
+      throws GenericEntityException, UsedIssueTypeException {
     WorkflowSchemeManager workflowSchemeManager = ComponentAccessor.getWorkflowSchemeManager();
     GenericValue workflowScheme = workflowSchemeManager.getWorkflowScheme(project);
-    workflowSchemeManager.addWorkflowToScheme(
-        workflowScheme, workflow.getName(), issueType.getId());
-  }
-
-  public static IssueType getLgtmIssueType() {
-    return getIssueTypeByName(Constants.ISSUE_TYPE_NAME);
+    Map<String, String> workflowMap = workflowSchemeManager.getWorkflowMap(project);
+    if (workflowMap.containsKey(issueType.getId())) {
+      if (!workflowMap.get(issueType.getId()).equals(workflow.getName())) {
```

Comment:
I'm not sure this achieves quite what we want (unless I'm missing something). If the `IssueType` has not been explicitly associated with a workflow scheme, perhaps it won't appear in the `workflowMap`, but nonetheless falls back to the `default` scheme?

Either way, trying this with existing tickets in a local Jira instance, I didn't get the required error when saving a new config.

Reply:
I thought if there are any existing tickets of type `LGTM alert` then we would need to raise this for manual intervention by the user?

Reply:
It only throws an error when the `LGTM alert` issue type is associated with a workflow that is not `LGTM alert`.

Reply:
What error are you missing?

Reply:
I was expecting an error on trying to save plugin config when I already had tickets of type `LGTM alert`.

Perhaps I'm wrong in expecting that to throw an error? What guarantees do we have about the workflow of existing tickets if this check passes?

Reply:
I think the `if` branch could be removed now, as the `else` branch should cover the dangerous case where issues already exist.